client/src/Components/Annotation/AnnotationViewer.tsx

```diff
@@ -25,24 +36,51 @@ const AnnotationViewer = (props) => {
     const annot_pos_y = e.target.id().split('-')[2];
     annotations[annot_indx][annot_pos_x] = x / spectrogramWidthScale; // reverse the calcs done to generate the coords
     annotations[annot_indx][annot_pos_y] = y;
-    forceUpdate(); // TODO remove the forceupdate and do it the proper way (possibly using spread?)
-
-    // Now update the actual meta.annotations
-    const f = annotations[annot_indx]['index']; // remember there are 2 different indexes- the ones on the screen and the meta.annotations
-    const updatedAnnotations = [...meta.annotations];
-    const updatedAnnotation = { ...updatedAnnotations[f] };
     const start_sample_index = lowerTile * TILE_SIZE_IN_IQ_SAMPLES;
-    updatedAnnotation['core:sample_start'] = annotations[annot_indx].y1 * fftSize * zoomLevel + start_sample_index;
-    updatedAnnotation['core:sample_count'] =
-      (annotations[annot_indx].y2 - annotations[annot_indx].y1) * fftSize * zoomLevel;
     const lower_freq = meta.captures[0]['core:frequency'] - meta.global['core:sample_rate'] / 2;
-    updatedAnnotation['core:freq_lower_edge'] =
+    const f = annotations[annot_indx]['index'];
+    meta.annotations[f]['core:sample_start'] = annotations[annot_indx].y1 * fftSize * zoomLevel + start_sample_index;
+    meta.annotations[f]['core:sample_count'] =
+      (annotations[annot_indx].y2 - annotations[annot_indx].y1) * fftSize * zoomLevel;
+    meta.annotations[f]['core:freq_lower_edge'] =
       (annotations[annot_indx].x1 / fftSize) * meta.global['core:sample_rate'] + lower_freq;
-    updatedAnnotation['core:freq_upper_edge'] =
+    meta.annotations[f]['core:freq_upper_edge'] =
       (annotations[annot_indx].x2 / fftSize) * meta.global['core:sample_rate'] + lower_freq;
-    dispatch(setMetaAnnotation({ annotation: updatedAnnotation, index: f }));
+    updateMeta(queryClient, meta);
+
+    // Now update the actual meta.annotations
```

Comment:
Is this commented code needed for something?

Reply:
THis is the code to do the update in the annotsations this need to be back before we can merge the PR without this the annostations will not work